pyproject.toml

```diff
@@ -65,6 +65,7 @@ all = [
     "canonicaljson",
     "jinja2",
     "jmespath",
+    "json_merge_patch",
```

Comment:
I believe this is a duplicate as `json_merge_patch` and `json-merge-patch` refer to the same pip package. We should remove one of them and adhere to the chosen naming convention.

https://github.com/bluesky/tiled/blob/8c717a3555dc4f129fd1bd20946b9e3ced7602bf/pyproject.toml#L71